apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -159,9 +159,9 @@ internal static IManagedTracer CreateManagedTracer(IServiceProvider provider)
                 string.Format(message, typeof(TraceProto).GetType()));
 
 
-            // TODO(talarico): Update this to allow header to not allow tracing.
             // If the trace header and rate limiter say not to trace, return a no-op tracer.
-            if (!headerContext.ShouldTrace && !rateLimitingFactory.CreateOptions().ShouldTrace)                
+            if (headerContext.ShouldTrace == false || (headerContext.ShouldTrace == null && 
```

Comment:
Correct.  This (and the below code) was the last piece of common code I need to pull out.  I have an action item to clean this up